src/test/java/net/logstash/logback/appender/LogstashTcpSocketAppenderTest.java

```diff
@@ -324,22 +346,12 @@ public void testReconnectToSecondaryOnWrite() throws Exception {
         appender.addDestination("localhost:10000");
         appender.addDestination("localhost:10001");
 
-        // Primary accepts first connection attempt then refuses
-        doNothing()
-            .doThrow(SocketTimeoutException.class)
-            .when(socket).connect(host("localhost", 10000), anyInt());
-
-        // Secondary refuses all attempts
-        doThrow(SocketTimeoutException.class)
-                .when(socket).connect(host("localhost", 10001), anyInt());
-
-        // First attempt of sending the event throws an exception while subsequent
-        // attempts will succeed. This should force the appender to close the connection
+        // First attempt at sending the event throws an exception while subsequent
+        // attempts succeed. This should force the appender to close the connection
         // and attempt to reconnect
-        doThrow(new RuntimeException())
-            .doReturn("event1".getBytes(StandardCharsets.UTF_8))
-            .when(encoder).encode(event1);
-        
+        doThrow(IOException.class)
+            .doNothing()
+            .when(outputStream).write(any(byte[].class), anyInt(), anyInt());
```

Comment:
It's unclear why this test needed to change for the refactoring of the executor service.  Can you clarify?

Reply:
Changing these tests was not actually required by the feature. 
The initial version hooks on the Encoder#encode() method to simulate the scenario. However, if the encoder is also a StreamingEncoder the streaming version of that method is invoked instead which makes the test brittle. In order to not make any assumption about the encoder (which is irrelevant in this test case) I decided to throw an exception from the output stream instead. This makes the test case more obvious (IMHO) and similar to the others already present in the same class...

Reply:
Ok.  Thanks for clarifying. :)

Just want to make sure the same scenario is being tested and verified.  I was worried because the stubs for the connect method were removed.  But after a deeper look, I think the verifications ensure that the same scenario is tested.

Reply:
I should have made these modifications in a separate PR... Honestly, it took me some time to remember what these tests are doing. I wrote them a couple of years ago already and after this second read I realised that some were not so obvious and could be made more robust. I hope they will be easier to read next time...